provider/provider.go

```diff
@@ -38,6 +38,12 @@ func Provider() terraform.ResourceProvider {
 				Sensitive:    true,
 				AtLeastOneOf: []string{"password", "auth_token"},
 			},
+			"api_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SIGSCI_URL", nil),
```

Comment:
Should this be an empty string `""`, rather than `nil`? 

We can't type assert an `interface{}` type containing a `nil` to a string ([example](https://go.dev/play/p/Mu-hVH3ujlW)).

So I'm not sure if this will work.

Reply:
@Integralist thanks for taking a look. 

I reused the schemas above this one which all use `nil` here (ex):

```
            "corp": {
                Type:        schema.TypeString,
                Required:    true,
                Description: "Corp short name (id)",
                DefaultFunc: schema.EnvDefaultFunc("SIGSCI_CORP", nil),
            },  
```

I think this is where the TF plugin is ensuring the return value isn't nil:

https://github.com/hashicorp/terraform-plugin-sdk/blob/v1.14.0/helper/schema/resource_data.go#L546-L548

I also printed it out just to confirm, and it's a string:

```
diff --git a/provider/provider.go b/provider/provider.go
index a93940f..f08c2f1 100644
--- a/provider/provider.go
+++ b/provider/provider.go
@@ -1,6 +1,8 @@
 package provider
 
 import (
+	"fmt"
+
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 	"github.com/signalsciences/go-sigsci"
@@ -72,6 +74,8 @@ func Provider() terraform.ResourceProvider {
 
 func providerConfigure() schema.ConfigureFunc {
 	return func(d *schema.ResourceData) (interface{}, error) {
+		fmt.Printf("auth_token: %#v\n", d.Get("auth_token"))
+		fmt.Printf("auth_token string conversion: %#v\n", d.Get("auth_token").(string))
 		config := Config{
 			Email:    d.Get("email").(string),
 			Password: d.Get("password").(string),
```

```
➜  terraform-provider-sigsci git:(config-set-url) ✗ make testacc
TF_ACC=1 go test -v ./...

...

auth_token: ""
auth_token string conversion: ""
```

They should probably all use `""` rather than `nil` anyway, though. Do you think we should update all of them to use `""`? Or keep it consistent for now and update them all in a follow-up PR?

Reply:
Yeah I think it makes the code clearer if they're all `""` rather than `nil`, and yeah let's do that as a separate PR 👍🏻 

But otherwise thanks for investigating and confirming 🙂